R/get_data.R

```diff
@@ -0,0 +1,467 @@
+#' Download species occurrence data from GBIF and prepare them for climate casting
+#'
+#' @description
+#' This function downloads species occurrence records from GBIF, filters them,
+#' and converts them into an sf data frame which can then be used for climate casting.
+#'
+#' @param taxon_key A numeric value or vector specifying the GBIF taxonKey(s).
+#' @param basis_of_record Optional character indicating the basisOfRecord types to be included in the data.
+#' If NULL, the default, occurrences with the following basisOfRecord will be kept:  "OBSERVATION", "HUMAN_OBSERVATION",
+#' "MATERIAL_SAMPLE", "LITERATURE", "PRESERVED_SPECIMEN", "UNKNOWN", and "MACHINE_OBSERVATION".
+#' @param coord_unc Optional numeric indicating the maximal coordinate uncertainty (m) an
+#' observation can have to be included in the data.
+#' If NULL, the default, all occurrences will be kept regardless of their coordinate uncertainty.
+#' @param identification_verification_status Optional character or a character vector indicating the identificationVerificationStatus of occurrence records that will be kept.
+#' If NULL, the default, all occurrences will be kept except those with the following identificationVerificationStatus: "unverified", "unvalidated", "not validated", "under validation", "not able to validate", "control could not be conclusive due to insufficient knowledge",  "Control could not be conclusive due to insufficient knowledge", "1","uncertain", "unconfirmed", "Douteux", "Invalide", "Non réalisable", "verification needed" , "Probable", "unconfirmed - not reviewed", "validation requested".
+#' @param region_shape Optional character specifying a path to a shape file indicating the region from which occurrence records should be downloaded.
+#'
+#' @returns An sf data frame holding occurrence records that are ready to be used for climate casting.
+#'
+#' @export
+#'
+#' @examples
+#'  \dontrun{
+#' #provide GBIF taxon_key(s)
+#' taxon_key <- c(2865504, 5274858)
+#' get_gbif_data(taxon_key)
```

Comment:
@soriadelva any reason you don't store the result of these steps in the environment ? ea add `df <-` ?